app/controllers/media_controller.rb

```diff
@@ -0,0 +1,84 @@
+class MediaController < ApplicationController
+  before_action only: [:show, :edit, :update, :destroy, :upvote] { @medium = Medium.find(params[:id]) }
+  before_action :set_type
+
+  def index
+    @media = type_class.all
+    @type = params[:type]
+  end
+
+  def show
+  end
+
+  def new
+    @medium = type_class.new
+    @type = @medium.type
+  end
+
+  def create
+    @medium = Medium.new(medium_params)
+      if @medium.save
+        redirect_to @medium
+      else
+        render action: 'new'
+      end
+  end
+
+  def edit
+    @type = @medium.type
+  end
+
+  def update
+    if @medium.update(medium_params)
+      redirect_to @medium
+    else
+      redirect_to media_path(@medium.type, @medium, :edit)
+    end
+  end
+
+  def destroy
+    type = @medium.type
+    @medium.destroy
+    if type == "Movie"
+      redirect_to movies_path
+    elsif type == "Book"
+      redirect_to books_path
+    elsif type == "Album"
+      redirect_to albums_path
+    else
+      redirect_to root
+    end
```

Comment:
Not sure if you've covered this in class or not, but since you're in the 'above and beyond' stage of trying single table inheritance, I'll suggest some extra things in this review.

You could consider using case switch statements here instead of such a long if/elsif/elsif/elsif/else statement. Here is a [quick description of them](http://www.skorks.com/2009/08/how-a-ruby-case-statement-works-and-what-you-can-do-with-it/)